### PR TITLE
Feat/load pretrained

### PR DESCRIPTION
**Pull request by galtimur**
Added script load_model_from_ckpt.py
Function `load_model_from_ckpt` takes the path to the checkpoint and returns AutoCompressorModel class with loaded checkpoint along with config on the run that was used to run this checkpoint.
Checkpoint parent folder should contain `base_model` folder with the state of the base model and the config of the run for this checkpoint.

The function reads from config_run whether the model was LORA or embeddings only. Then loads the model. Code for LLAMA loading I took from @mu-arkhipov function `merge_save_lora`

Load only embeddings function merges this base model with checkpoint state, creating `checkpoint_merge_temp` folder.
See for example files in `/mnt/data2/galimzyanov/autocompressor/checkpoints/LLaMA-1.3B_sub3_seg2_sum50_embed_only_test` folder

UPD. [e881b71] adds loading of the tokenizer

### CODE REVIEW

**Comment by mu-arkhipov** [comment]
Just noticed that changes in configs are removed from PR, that's nice!

File: configs/config.yaml [hunk 1]
```diff
@@ -11,7 +11,7 @@ defaults:
   logging_steps: 1
   add_special_tokens: false
   use_fast_tokenizer: false
-  num_gpus: 4
+  num_gpus: 2
```
Comment (mshnmshn) [inline configs/config.yaml]: Do I need to change `num_gpus` when I need other than 2 gpus?

File: configs/config.yaml [hunk 2]
```diff
@@ -52,7 +52,7 @@ training:
   randomize_substeps: true
   segment_gradient_checkpointing: false
   learning_rate: 4e-3
-  resume_from_checkpoint: false
+  resume_from_checkpoint: true
```
Comment (mshnmshn) [inline configs/config.yaml:53]: Does `resume_from_checkpoint: true` work when there is no checkpoints?

File: config_parser.py [hunk 3]
```diff
@@ -37,7 +37,7 @@ def parse_config(config_path):
     #     run_name_suffix += "_check"
     if config["train_embed_only"]:
         run_name_suffix += "_embed_only"
-    # run_name_suffix += "_test"
+    run_name_suffix += "_test"
```
Comment (mshnmshn) [inline config_parser.py:41]: May be leave it commented?

File: configs/accelerate_2gpu.yaml [hunk 4]
```diff
@@ -2,7 +2,7 @@ compute_environment: LOCAL_MACHINE
 debug: false
 distributed_type: MULTI_GPU
 downcast_bf16: 'no'
-gpu_ids: "0,1"
+gpu_ids: "0,2"
```
Comment (mshnmshn) [inline configs/accelerate_2gpu.yaml]: Want to leave it as 0,1

File: train.py [hunk 5]
```diff
@@ -21,6 +21,12 @@
 from base_trainer import EvalCallback
 from utils import get_last_checkpoint_or_last_model, parse_checkpoint_step, load_check_merging
 from config_parser import parse_config
+import shutil
+from pathlib import Path
+
+from accelerate import Accelerator
+
+accelerator = Accelerator()
```
Comment (mshnmshn) [inline train.py]: Do we need `gradient_accumulation_steps` here? Like `accelerator = Accelerator(gradient_accumulation_steps=2)` (docs link https://huggingface.co/docs/accelerate/en/package_reference/accelerator#gradient-accumulation)

File: configs/config.yaml (hunk 1 above)
Comment (galtimur) [reply configs/config.yaml]: That's the problem I do not understand how to solve. `num_gpus` is needed only to calculate a number of accumulation steps per GPU, since in config we set total batch accum and minibatch size per device. But accelerate config needs `num_gpus` to set up real number of devices.

File: config_parser.py (hunk 3 above)
Comment (galtimur) [reply config_parser.py:41]: Sure!

File: configs/accelerate_2gpu.yaml (hunk 4 above)
Comment (galtimur) [reply configs/accelerate_2gpu.yaml]: Not so easy - it depends on what GPUs are available. But probably I should control it by CUDA_VISIBLE_DEVICES. I would change it.

File: train.py (hunk 5 above)
Comment (galtimur) [reply train.py]: Actually, this import and declaration is redundant. I forgot to delete it. I do not use it, since, HF Trainer initializes accelerate itself.

File: configs/config.yaml (hunk 2 above)
Comment (galtimur) [reply configs/config.yaml:53]: It tries to find checkpoint in out folder first. If there is no such, then prints

`f"Didn't find a checkpoint in {training_args.output_dir}. Starting training from scratch"`

And starts from the scratch.

File: configs/config.yaml (hunk 1 above)
Comment (galtimur) [reply configs/config.yaml]: I fixed this issue in [2effd4b] commit. Removed n_gpus and n_nodes from config. Now it takes this information from accelerate.

File: load_model_from_ckpt.py [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+    elif merged_config["train_embed_only"]:
+        model, tokenizer = load_only_embed_model_from_ckpt(checkpoint_path, merged_config)
+        print("Loaded embeddings only model")
+    else:
+        print("Loading for this config state is not implemented yet")
+
+    return model, tokenizer, merged_config
+
+# checkpoint_path = "/mnt/data2/galimzyanov/autocompressor/checkpoints/LLaMA-1.3B_sub3_seg2_sum50_embed_only_test/checkpoint-9900"
+checkpoint_path = "/mnt/data2/galimzyanov/autocompressor/checkpoints/LLaMA-1.3B_sub3_seg2_sum50/checkpoint-2250"
```
Comment (mshnmshn) [inline load_model_from_ckpt.py]: Want to add `if __name__ == '__main__'` here

Comment (mshnmshn) [reply load_model_from_ckpt.py]: Added

**Review by mshnmshn — APPROVED** [review]
